src/js/components/modals/ServiceFormModal.js

```diff
@@ -258,13 +270,15 @@ class ServiceFormModal extends mixin(StoreMixin) {
     }
 
     if (this.state.jsonMode) {
-      let jsonDefinition = this.state.jsonDefinition;
-      marathonAction(JSON.parse(jsonDefinition), this.state.force);
+      let jsonDefinition = JSON.parse(this.state.jsonDefinition);
+      jsonDefinition = cleanJSONdefinition(jsonDefinition);
+      marathonAction(jsonDefinition, this.state.force);
+      jsonDefinition = JSON.stringify(jsonDefinition, null, 2);
```

Comment:
Isn't this a bit convoluted?
1) parse the json string
2) remove fields from it and convert to object
3) (call the action)
4) re-convert parsed json back to string
...